homeassistant/components/ccm15/coordinator.py

```diff
@@ -0,0 +1,332 @@
+        """Test the connection to the CCM15 device."""
+        url = f"http://{self._host}:{self._port}/{CONF_URL_STATUS}"
+        try:
+            async with aiohttp.ClientSession() as session, session.get(
+                url, timeout=10
+            ) as response:
+                if response.status == 200:
+                    return True
+                _LOGGER.debug("Test connection: Cannot connect : %s", response.status)
+                return False
+        except (aiohttp.ClientError, asyncio.TimeoutError):
+            _LOGGER.debug("Test connection: Timeout")
+            return False
+
+    async def async_send_state(self, url: str) -> bool:  # pragma: no cover
+        """Send the url to set state to the ccm15 slave."""
+        async with httpx.AsyncClient() as client:
+            response = await client.get(url, timeout=DEFAULT_TIMEOUT)
+            _LOGGER.debug("API response status code:%d", response.status_code)
+            return response.status_code in (httpx.codes.OK, httpx.codes.FOUND)
+
+    async def async_set_state(self, ac_index: int, state: str, value: int) -> None:
+        """Set new target states."""
+        _LOGGER.debug("Calling async_set_states for ac index '%s'", ac_index)
+        ac_id: int = 2**ac_index
+        url = BASE_URL.format(
+            self._host,
+            self._port,
+            CONF_URL_CTRL
+            + "?ac0="
+            + str(ac_id)
+            + "&ac1=0"
+            + "&"
+            + state
+            + "="
+            + str(value),
+        )
+        _LOGGER.debug("Url:'%s'", url)
+
+        if await self.async_send_state(url):
+            await self.async_request_refresh()
+
+    def get_ac_data(self, ac_index: int) -> Optional[CCM15SlaveDevice]:
+        """Get ac data from the ac_index."""
+        if ac_index < 0 or ac_index >= len(self.data.devices):
+            # Index is out of bounds or not an integer
+            _LOGGER.warning(
+                "Data is not available for AC id %s.%s", self._host, ac_index
+            )
+            return None
+        data = self.data.devices[ac_index]
+        return data
+
+    async def async_set_hvac_mode(self, ac_index, hvac_mode: HVACMode) -> None:
+        """Set the hvac mode."""
+        _LOGGER.debug("Set Hvac[%s]='%s'", ac_index, str(hvac_mode))
+        await self.async_set_state(ac_index, "mode", CONST_STATE_CMD_MAP[hvac_mode])
+
+    async def async_set_fan_mode(self, ac_index, fan_mode: str) -> None:
+        """Set the fan mode."""
+        _LOGGER.debug("Set Fan[%s]='%s'", ac_index, fan_mode)
+        await self.async_set_state(ac_index, "fan", CONST_FAN_CMD_MAP[fan_mode])
+
+    async def async_set_temperature(self, ac_index, temp) -> None:
+        """Set the target temperature mode."""
+        _LOGGER.debug("Set Temp[%s]='%s'", ac_index, temp)
+        await self.async_set_state(ac_index, "temp", temp)
+
+
+class CCM15Climate(CoordinatorEntity[CCM15Coordinator], ClimateEntity):
```

Comment:
New integrations need to add support for [translating name and attributes](https://developers.home-assistant.io/blog/2023/03/27/entity_name_translations/)

Reply:
"temp" is not a user visible name, it is the actual parameter that must be used to set target temperature.